src/graph/ChcGraph.cc

```diff
@@ -599,4 +601,126 @@ std::unique_ptr<ChcDirectedHyperGraph> ChcDirectedHyperGraph::makeEmpty(Logic &
+        }
+        return ret;
+    }();
+
+    vec<SRef> argSorts;
+    std::vector<PTRef> vars;
+    for (auto arg : stateVars) {
+        argSorts.push(logic.getSortRef(arg));
+        vars.push_back(timeMachine.getUnversioned(arg));
+    }
+
+    PTRef transitionRelation = logic.mkOr(std::move(transitionRelationComponent));
+    std::string name;
+    for (auto vertex : vertices) {
+        name += logic.getSymName(vertex);
+    }
+
+    SymRef newRef = logic.declareFun(name, logic.getSort_bool(), argSorts);
+    predicates.addRepresentation(newRef, std::move(vars));
+
+    // KB: Changing input and output edges to and from the loop in the graph with new state and loc variables
+    //     Redirecting edges to the newly created vertice
+    allEdges = getEdges();
+    for (auto edge : allEdges) {
+        if (std::find(vertices.begin(), vertices.end(), getSource(edge)) != vertices.end()) {
+            PTRef newSourceLoc = locationVars.at(getSource(edge));
+            auto oldEdgeVars = getVariablesFromEdge(logic, *this, edge);
+            std::unordered_map<PTRef, PTRef, PTRefHash> subMap;
+            for (uint i = 0; i < oldEdgeVars.stateVars.size(); i++) {
+                subMap.insert(std::make_pair(oldEdgeVars.stateVars[i], argVars[{getSource(edge), i}]));
+            }
+            updateEdgeSource(edge, newRef);
+            PTRef label = utils.varSubstitute(logic.mkAnd(getEdgeLabel(edge), newSourceLoc), subMap);
+
+            updateEdgeLabel(edge, InterpretedFla{label});
+        }
+        if (std::find(vertices.begin(), vertices.end(), getTarget(edge)) != vertices.end()) {
+
+            PTRef newTargetLoc = [&]() -> PTRef {
+                vec<PTRef> negatedLocations;
+                negatedLocations.capacity(locationVars.size());
+                for (auto && entry : locationVars) {
+                    if (entry.first != getTarget(edge)) {
+                        negatedLocations.push(logic.mkNot(timeMachine.sendVarThroughTime(entry.second, 1)));
+                    }
+                }
+                return logic.mkAnd(std::move(negatedLocations));
+            }();
+            PTRef destination = timeMachine.sendVarThroughTime(locationVars.at(getTarget(edge)), 1);
+
+            auto oldEdgeVars = getVariablesFromEdge(logic, *this, edge);
+            std::unordered_map<PTRef, PTRef, PTRefHash> subMap;
+            for (uint i = 0; i < oldEdgeVars.nextStateVars.size(); i++) {
+                subMap.insert(std::make_pair(oldEdgeVars.nextStateVars[i],
+                                             timeMachine.sendVarThroughTime(argVars[{getTarget(edge), i}], 1)));
+            }
+            updateEdgeTarget(edge, newRef);
+            PTRef label = utils.varSubstitute(getEdgeLabel(edge), subMap);
+
+            updateEdgeLabel(edge, InterpretedFla{rewriteMaxArityClassic(
+                                            logic, logic.mkAnd(logic.mkAnd(newTargetLoc, destination), label))});
```

Comment:
```suggestion
            updateEdgeLabel(edge, InterpretedFla{logic.mkAnd({newTargetLoc, destination, label})});
```
This should be enough, no?

Reply:
Fixed